### PR TITLE
add support for detecting 45GS02

### PR DESCRIPTION
**Pull request by gardners**
Added support for detecting for MEGA65 45GS02 CPU.
Also renamed a bunch of labels to make it easier to follow the flow of getcpu().

### CODE REVIEW

File: libsrc/common/getcpu.s [hunk 1]
```diff
@@ -26,52 +27,118 @@
...
 
         .byte   $42,$EA         ; neg on 65CE02/4510, nop #$EA on 65C02, wdm $EA on 65816
         cmp     #1
-        beq     @L6
+        beq     @HasINCA
 
 ; This is at least a 65CE02, check for 4510
 
         lda     #5              ; CPU_65CE02 constant
+	ldx	#0		; To make sure MAP doesn't do anything, the upper nybl of X and Z must be clear
```
Comment (oliverschmidt) [inline libsrc/common/getcpu.s]: Please remove the tab char.

File: libsrc/common/getcpu.s [hunk 2]
```diff
@@ -26,52 +27,118 @@
...
         .byte   $5C             ; map on 4510, aug on 65CE02 (acts like 4 byte nop)
         lda     #3              ; CPU_4510 constant
         nop
-        bne     @L9
+	cmp	#5
```
Comment (oliverschmidt) [inline libsrc/common/getcpu.s]: Please remove the tab char.

File: libsrc/common/getcpu.s [hunk 3]
```diff
@@ -26,52 +27,118 @@
...
+        ; now try again to load it: If the same, then 45GS02, as $200FF is unchanged
+        nop                     ; prefix to tell next instruction to be 32-bit ZP
+        .byte   $b2,$fb         ; LDA ($nn),Z
+        cmp     $ff             ; does the loaded value match what is in $FF?
+        beq     @Is4510         ; matches, so must be a 4510 = C65
+        bne     @Is45GS02       ; $200FF and $FF have different values, so must be a MEGA65 45GS02
+@Is4510:
+        jsr     @RestoreGetCPUTemp
+        lda     #$03            ; CPU_4510 constant
+        ldx     #0              ; Load high byte of word
```
Comment (oliverschmidt) [inline libsrc/common/getcpu.s]: Please decide on #0 vs. #$0 vs. #$00 coding style and be consistent about it.

Comment (oliverschmidt) [reply libsrc/common/getcpu.s]: Please decide on starting comments with upper case letter or not and be consistent about it.

File: libsrc/common/getcpu.s [hunk 4]
```diff
@@ -26,52 +27,118 @@
...
+        ; Save the 32-bit ZP pointer and data byte
+        ldx     #$04
+@L11:   lda     @GetCPUTemp,x
+        sta     $fb,x
+        dex
+        bpl     @L11
+        rts
+
+        ; Temporary storage for the ZP locations we modify above
+@GetCPUTemp:    .byte   0,0,0,0,0
```
Comment (oliverschmidt) [inline libsrc/common/getcpu.s]: This isn't code is it doesn't belong in the `.code´ section. Please inform yourself a bit about the role of segments and their usage in ca65. Hint: To me this looks like it "wants" to read something like
```
        .bss

@GetCPUTemp:
        .res 5

        .code
```

File: libsrc/common/getcpu.s [hunk 5]
```diff
@@ -26,52 +27,118 @@
...
-@L4:    ldx     #6              ; CPU_HUC6280 constant
+@CheckHuC6280:    ldx     #6              ; CPU_HUC6280 constant
         .byte   $22,$EA         ; sax nop on HuC6280 (A=$06, X=$01), nop #$EA on 65C02 (A=$01, X=$06)
-        bne     @L9
+        bne     @LoadXAndReturn
 
 ; Check for 65816/65802
-@L6:    xba                     ; .byte $EB, put $01 in B accu (nop on 65C02/65SC02)
+@HasINCA:
+	xba                     ; .byte $EB, put $01 in B accu (nop on 65C02/65SC02)
```
Comment (oliverschmidt) [inline libsrc/common/getcpu.s]: Please remove the tab char.

File: libsrc/common/getcpu.s [hunk 6]
```diff
@@ -82,9 +149,8 @@ _getcpu:
...
-        bne     @L4
+        bne     @CheckHuC6280
         lda     #4              ; CPU_65SC02 constant
-
-@L9:    ldx     #0              ; Load high byte of word
-        rts
-
+@LoadXAndReturn:
+	ldx	#0
+	rts
```
Comment (oliverschmidt) [inline libsrc/common/getcpu.s]: Please remove the tab chars.

**Comment by gardners** [comment]
Thanks for the quick feedback. I have made all the requested changes by amending the commit.  I'll thus close this PR and create a fresh one with the amended commit to keep things simple.

**Comment by oliverschmidt** [comment]
Thanks for your quick reaction - and you intention to keep things simple. However, It's the very idea of pull request to have a history. So please do *NOT* close a pr and create a new one. Instead just push additional commits to the Git branch in question and post a quick comment in the pr conversation that you have done so.

**Comment by gardners** [comment]
ok. noted for next time.  In terms of squashing costmetic changes as in this case, is there as simple way to get the PR to shift to the amended commit, or do you prefer to just have the extra commits in the history?

**Comment by greg-king5** [comment]
> OK, noted for next time. In terms of squashing cosmetic changes, as in this case, is there a simple way to get the PR to shift to the amended commit; or, do you prefer to just have the extra commits in the history?

If you are using a command-line interface to Git, then you can use commands like this:
```sh
git push --force-with-lease <remote-repository> master
```
If you're using a GUI, then there should be some way to do the same thing (maybe a "force with lease" checkbox that you must click before pushing the amended commit).

**Pull request closed without merging**